examples/rvc-app/rvc-common/src/rvc-service-area-delegate.cpp

```diff
@@ -0,0 +1,415 @@
+        DataModel::Nullable<ServiceArea::FloorSurfaceTag>());
+
+    // Location C has full SemData, no name, Map YY
+    GetInstance()->AddSupportedLocation(supportedLocationId_C, DataModel::Nullable<uint_fast8_t>(supportedMapId_YY), CharSpan(),
+                                        DataModel::Nullable<int16_t>(-1),
+                                        DataModel::Nullable<ServiceArea::AreaTypeTag>(ServiceArea::AreaTypeTag::kPlayRoom),
+                                        DataModel::Nullable<ServiceArea::LandmarkTag>(ServiceArea::LandmarkTag::kBackDoor),
+                                        DataModel::Nullable<ServiceArea::PositionTag>(ServiceArea::PositionTag::kNextTo),
+                                        DataModel::Nullable<ServiceArea::FloorSurfaceTag>(ServiceArea::FloorSurfaceTag::kConcrete));
+
+    // Location D has null values for all HomeLocationStruct fields, Map YY
+    GetInstance()->AddSupportedLocation(supportedLocationId_D, DataModel::Nullable<uint_fast8_t>(supportedMapId_YY),
+                                        "My Location D"_span, DataModel::Nullable<int16_t>(),
+                                        DataModel::Nullable<ServiceArea::AreaTypeTag>(),
+                                        DataModel::Nullable<ServiceArea::LandmarkTag>(ServiceArea::LandmarkTag::kCouch),
+                                        DataModel::Nullable<ServiceArea::PositionTag>(ServiceArea::PositionTag::kNextTo),
+                                        DataModel::Nullable<ServiceArea::FloorSurfaceTag>(ServiceArea::FloorSurfaceTag::kHardwood));
+
+    GetInstance()->SetCurrentLocation(supportedLocationId_C);
+
+    return CHIP_NO_ERROR;
+}
+
+//*************************************************************************
+// command support
+
+bool RvcServiceAreaDelegate::IsSetSelectedLocationsAllowed(MutableCharSpan statusText)
+{
+    // TODO IMPLEMENT
+    return true;
+};
+
+bool RvcServiceAreaDelegate::IsValidSelectLocationsSet(const Commands::SelectLocations::DecodableType & req,
+                                                       SelectLocationsStatus & locationStatus, MutableCharSpan statusText)
+{
+    // TODO IMPLEMENT
+    return true;
+};
+
+bool RvcServiceAreaDelegate::HandleSkipCurrentLocation(MutableCharSpan skipStatusText)
+{
+    // TODO IMPLEMENT
+    return true;
+};
+
+//*************************************************************************
+// Supported Locations accessors
+
+bool RvcServiceAreaDelegate::IsSupportedLocationsChangeAllowed()
+{
+    // TODO IMPLEMENT
+    return true;
+}
+
+uint32_t RvcServiceAreaDelegate::GetNumberOfSupportedLocations()
+{
+    return static_cast<uint32_t>(mSupportedLocations.size());
+}
+
+bool RvcServiceAreaDelegate::GetSupportedLocationByIndex(uint32_t listIndex, LocationStructureWrapper & aSupportedLocation)
+{
+    if (listIndex < static_cast<uint32_t>(mSupportedLocations.size()))
```

Comment:
Again: why is this cast needed at all?

Reply:
Fixed [here](https://github.com/project-chip/connectedhomeip/pull/33991/commits/87646aaae78294fe999f0491b3e5a91ec58adb9c).